services/QuillLMS/spec/services/diagnostic_insights/skill_performance_by_classroom_spec.rb

```diff
@@ -174,5 +174,27 @@ module DiagnosticInsights
 
       it { expect(subject.first[described_class::NO_DATA].first['user_id']).to eq(user2.id) }
     end
+
+    context 'sorting skill order in response' do
+      let(:intended_order) { [2, 1] }
+      let(:skill_groups) { intended_order.map { |order_number| create(:skill_group, order_number:) } }
+      let!(:skill_group_activities) { skill_groups.map { |skill_group| create(:skill_group_activity, skill_group:, activity: pre_diagnostic) } }
+      let(:data) do
+        {
+          'skill_performance' => skill_groups.map do |skill_group|
+            {
+              skill_id: skill_group.id,
+              skill_name: skill_group.name,
+              questions_correct:,
+              questions_total:
+            }
+          end
+        }
+      end
+
+      let(:expected_order) { skill_groups.sort_by(&:order_number).map(&:id) }
+
+      it { expect(subject.map { |skill| skill['skill_id'] }).to eq(expected_order) }
```

Comment:
I like `pluck` for direct access to an attribute within a collection:
`it { expect(subject.pluck('skill_id')).to eq(expected_order) }`